src/ngResource/resource.js

```diff
@@ -568,17 +570,13 @@ angular.module('ngResource', ['ng']).
               undefined;
 
             forEach(action, function(value, key) {
-              switch (key) {
-                default:
-                  httpConfig[key] = copy(value);
-                  break;
-                case 'params':
-                case 'isArray':
-                case 'interceptor':
-                  break;
-                case 'timeout':
-                  httpConfig[key] = value;
-                  break;
+              if ((key === 'timeout') && value && !angular.isNumber(value)) {
+                $log.debug('ngResource:\n' +
+                           '  Only numeric values are allowed as `timeout`.\n' +
+                           '  Promises are not supported in $resource, because the same value ' +
+                           'has to be re-used for multiple requests.');
+              } else if (key != 'params' && key != 'isArray' && key != 'interceptor') {
+                httpConfig[key] = copy(value);
```

Comment:
I think I find the switch statement easier to follow. What is the benefit of changing to if?